src/Validation/ValidationRule.php

```diff
@@ -30,67 +30,24 @@
 class ValidationRule
 {
     /**
-     * The method to be called for a given scope
+     * The callable to be used for validation
      *
-     * @var callable|string
+     * @var callable
      */
-    protected $_rule;
-
-    /**
-     * The 'on' key
-     *
-     * @var callable|string|null
-     */
-    protected $_on;
-
-    /**
-     * The 'last' key
-     *
-     * @var bool
-     */
-    protected bool $_last = false;
-
-    /**
-     * The 'message' key
-     *
-     * @var string|null
-     */
-    protected ?string $_message = null;
-
-    /**
-     * Key under which the object or class where the method to be used for
-     * validation will be found
-     *
-     * @var string
-     */
-    protected string $_provider = 'default';
-
-    /**
-     * Extra arguments to be passed to the validation method
-     *
-     * @var array
-     */
-    protected array $_pass = [];
+    protected $callable;
 
     /**
      * Constructor
-     *
-     * @param array<string, mixed> $validator The validator properties
-     */
-    public function __construct(array $validator)
-    {
-        $this->_addValidatorProps($validator);
-    }
-
-    /**
-     * Returns whether this rule should break validation process for associated field
-     * after it fails
-     *
-     * @return bool
      */
-    public function isLast(): bool
-    {
-        return $this->_last;
+    public function __construct(
+        callable $callable,
+        public protected(set) ?string $name = null,
```

Comment:
Does any one have an option on using asymmetric visibility vs making the property readonly. The benefit of the former is the variable could be updated later within the class context if required (thought currently none of the props are modified after object construction).

Reply:
Done